### PR TITLE
Arm build support

### PR DESCRIPTION
**Pull request by ibrokethecloud**
**Problem:**

PR introduces experimental support for arm hosts

**Solution:**

**Related Issue:**
https://github.com/harvester/harvester/issues/371
**Test plan:**

### CODE REVIEW

**Comment by ibrokethecloud** [comment]
Additional work is still needed to identify correct arm based kubevirt images from registry.opensuse.org

The build currently patches harvester chart to download images from `quay.io/kubevirt` and packages the same with harvester.

File: scripts/build-bundle [hunk 1]
```diff
@@ -58,6 +59,15 @@ ${SCRIPTS_DIR}/patch-harvester ${TOP_DIR}/../harvester
 # Package harvester chart
 harvester_chart_path=${harvester_path}/deploy/charts/harvester
 harvester_crd_chart_path=${harvester_path}/deploy/charts/harvester-crd
+
+## for arm based builds we need to patch registry to registry.opensuse.com
```
Comment (tserong) [inline scripts/build-bundle:63]: Comment needs updating (`patch_harvester_chart` is using quay.io ATM)

File: scripts/package-harvester-os [hunk 2]
```diff
@@ -18,6 +18,7 @@ source ${SCRIPTS_DIR}/version-monitoring
 source ${SCRIPTS_DIR}/version-logging
 source ${SCRIPTS_DIR}/lib/iso
 
+## Revert back to sle-micro-head
```
Comment (tserong) [inline scripts/package-harvester-os:21]: Spurious comment?

**Review by tserong — APPROVED** [review]
LGTM - all I have is comments about comments ;-)

**Comment by bk201** [comment]
@ibrokethecloud `set -ex` is turn on in multiple scripts, do we need to be verbose?

**Review by Vicente-Cheng — APPROVED** [review]
LGTM.

There is just one point I would like to raise.

There are too many customized operations with different arch.
We might refine the iso building flow in the next version.

Thanks!

**Comment by Vicente-Cheng** [comment]
BTW, could we squash some commits to make the tracking easier?

**Review by bk201 — APPROVED** [review]

**Pull request merged**